ONSdigital/django-oauth2-test
# Enable logging in cloudfoundry

# Motivation and Context
No logs are output in cloudfoundry at the moment except from the route
logs. This change makes stdout logging default behaviour.

Also adding exception logging to the custom error handler.

Logging before
```
   2018-07-05T15:17:20.70+0100 [RTR/0] OUT django-oauth2-test-ci.apps.devtest.onsclofo.uk - [2018-07-05T14:17:20.029+0000] "POST /api/v1/tokens/ HTTP/1.1" 201 126 170 "-" "python-requests/2.18.4" "10.0.0.148:10530" "10.0.1.11:60956" x_forwarded_for:"34.252.57.26, 10.0.0.148" x_forwarded_proto:"http" vcap_request_id:"cb66353a-4e49-4202-5734-b6967c38ba77" response_time:0.680024885 app_id:"29c8bc52-d06a-4413-84cb-9561040bdf36" app_index:"0" x_b3_traceid:"9d3d9c5a1d124093" x_b3_spanid:"9d3d9c5a1d124093" x_b3_parentspanid:"-"
   2018-07-05T15:17:20.70+0100 [RTR/0] OUT
   2018-07-05T15:17:22.32+0100 [RTR/1] OUT django-oauth2-test-ci.apps.devtest.onsclofo.uk - [2018-07-05T14:17:21.705+0000] "POST /api/account/create HTTP/1.1" 201 106 74 "-" "python-requests/2.17.3" "10.0.5.121:13862" "10.0.1.11:60956" x_forwarded_for:"34.252.57.26, 10.0.5.121" x_forwarded_proto:"http" vcap_request_id:"ea85e121-9844-45ba-62db-ed110cdf5f75" response_time:0.621935703 app_id:"29c8bc52-d06a-4413-84cb-9561040bdf36" app_index:"0" x_b3_traceid:"f9851978117a60ce" x_b3_spanid:"f9851978117a60ce" x_b3_parentspanid:"-"
   2018-07-05T15:17:22.32+0100 [RTR/1] OUT
   2018-07-05T15:17:22.73+0100 [RTR/0] OUT django-oauth2-test-ci.apps.devtest.onsclofo.uk - [2018-07-05T14:17:22.404+0000] "PUT /api/account/create HTTP/1.1" 201 110 74 "-" "python-requests/2.18.4" "10.0.5.121:17904" "10.0.1.11:60956" x_forwarded_for:"52.215.90.63, 10.0.5.121" x_forwarded_proto:"http" vcap_request_id:"4a236cd7-f39d-4e80-5a3f-ffbc6ed7a00f" response_time:0.331040176 app_id:"29c8bc52-d06a-4413-84cb-9561040bdf36" app_index:"0" x_b3_traceid:"ed3a2b5337cb773b" x_b3_spanid:"ed3a2b5337cb773b" x_b3_parentspanid:"-"
   2018-07-05T15:17:22.73+0100 [RTR/0] OUT
   2018-07-05T15:17:25.99+0100 [RTR/1] OUT django-oauth2-test-ci.apps.devtest.onsclofo.uk - [2018-07-05T14:17:25.355+0000] "POST /api/account/create HTTP/1.1" 201 106 74 "-" "python-requests/2.17.3" "10.0.5.121:14198" "10.0.1.11:60956" x_forwarded_for:"34.252.57.26, 10.0.5.121" x_forwarded_proto:"http" vcap_request_id:"71d542db-8fb5-4f64-41a1-598ab417856f" response_time:0.640804105 app_id:"29c8bc52-d06a-4413-84cb-9561040bdf36" app_index:"0" x_b3_traceid:"863b251e343f2102" x_b3_spanid:"863b251e343f2102" x_b3_parentspanid:"-"
   2018-07-05T15:17:25.99+0100 [RTR/1] OUT
   2018-07-05T15:17:26.38+0100 [RTR/1] OUT django-oauth2-test-ci.apps.devtest.onsclofo.uk - [2018-07-05T14:17:26.051+0000] "PUT /api/account/create HTTP/1.1" 201 110 74 "-" "python-requests/2.18.4" "10.0.5.121:13862" "10.0.1.11:60956" x_forwarded_for:"52.215.90.63, 10.0.5.121" x_forwarded_proto:"http" vcap_request_id:"592f0d62-da5a-4588-6821-46fcf6333410" response_time:0.333838842 app_id:"29c8bc52-d06a-4413-84cb-9561040bdf36" app_index:"0" x_b3_traceid:"5181210f01dbe5cf" x_b3_spanid:"5181210f01dbe5cf" x_b3_parentspanid:"-"
   2018-07-05T15:17:26.38+0100 [RTR/1] OUT
   2018-07-05T15:17:34.18+0100 [RTR/0] OUT django-oauth2-test-ci.apps.devtest.onsclofo.uk - [2018-07-05T14:17:33.462+0000] "POST /api/v1/tokens/ HTTP/1.1" 201 126 170 "-" "python-requests/2.18.4" "10.0.0.148:7610" "10.0.1.11:60956" x_forwarded_for:"34.252.57.26, 10.0.0.148" x_forwarded_proto:"http" vcap_request_id:"d453e33a-6d35-4b14-5f56-cab842dcbe43" response_time:0.717402126 app_id:"29c8bc52-d06a-4413-84cb-9561040bdf36" app_index:"0" x_b3_traceid:"a7ca79d5c4a4f1dc" x_b3_spanid:"a7ca79d5c4a4f1dc" x_b3_parentspanid:"-"
   2018-07-05T15:17:34.18+0100 [RTR/0] OUT
```

Logging after
```
  2018-07-05T15:26:10.21+0100 [APP/PROC/WEB/0] OUT  {"created":"2018-07-05 14:26:10 ", "service":"oauth2", "level": "DEBUG", "event":"Token validate request decorator being called", "context": "apps.tokens.decorators.validate_request:113"  }
   2018-07-05T15:26:10.21+0100 [APP/PROC/WEB/0] OUT  {"created":"2018-07-05 14:26:10 ", "service":"oauth2", "level": "DEBUG", "event":"Decorator called for Validating OAuth2 requests", "context": "apps.tokens.decorators.decorator:336"  }
   2018-07-05T15:26:10.21+0100 [APP/PROC/WEB/0] OUT  {"created":"2018-07-05 14:26:10 ", "service":"oauth2", "level": "DEBUG", "event":"Grant_type password", "context": "apps.tokens.decorators._validate_grant_type:157"  }
   2018-07-05T15:26:10.21+0100 [APP/PROC/WEB/0] OUT  {"created":"2018-07-05 14:26:10 ", "service":"oauth2", "level": "DEBUG", "event":"Grant_type password. Second function", "context": "apps.tokens.decorators._validate_grant_type:215"  }
   2018-07-05T15:26:10.51+0100 [APP/PROC/WEB/0] OUT  {"created":"2018-07-05 14:26:10 ", "service":"oauth2", "level": "DEBUG", "event":"*** We have a HTTP_AUTHORIZATION request ***", "context": "apps.tokens.decorators._extract_client:267"  }
   2018-07-05T15:26:10.79+0100 [APP/PROC/WEB/0] OUT  {"created":"2018-07-05 14:26:10 ", "service":"oauth2", "level": "DEBUG", "event":"Hitting post TokensView", "context": "apps.tokens.views.post:17"  }
   2018-07-05T15:26:10.80+0100 [RTR/1] OUT django-oauth2-test-ci.apps.devtest.onsclofo.uk - [2018-07-05T14:26:10.118+0000] "POST /api/v1/tokens/ HTTP/1.1" 201 115 170 "-" "python-requests/2.18.4" "10.0.5.121:62922" "10.0.1.11:60960" x_forwarded_for:"34.252.57.26, 10.0.5.121" x_forwarded_proto:"http" vcap_request_id:"5cc04d18-f5ec-4520-640e-2316f0586840" response_time:0.688014798 app_id:"29c8bc52-d06a-4413-84cb-9561040bdf36" app_index:"0" x_b3_traceid:"d4f4a86acd6b4e4d" x_b3_spanid:"d4f4a86acd6b4e4d" x_b3_parentspanid:"-"
   2018-07-05T15:26:10.80+0100 [RTR/1] OUT

```

# What has changed
Add logger.exception to generic error handler
Simplify the logging profiles
Fix logging to stdout

# How to test?
1. Check acceptance tests pass
1.
	1. Add `raise Exception` to https://github.com/ONSdigital/django-oauth2-test/blob/master/oauth2server/apps/accounts/views.py#L33
	1. Start service `OAUTH2_SUPER_USER=test OAUTH2_SUPER_USER_PASSWORD=test OAUTH2_SUPER_USER_EMAIL=test@test.com  pipenv run ./start.sh`
	1. Hit `http -f POST http://localhost:8040/api/account/create 'client_id=ons@ons.gov' 'client_secret=password' 'username=test@test.com' 'password=test'` (using https://httpie.org/ or curl)
